desktop/desktop
# Generate deltas

Fixes #51 

Some notable notes about the changes here:

1. We bake the update channel (e.g., `beta` or `production`) into the app at build time.
2. The Windows exe name is now `GitHubWindows` (no space). This is because Squirrel.Windows can't diff files with spaces in their names, so it'd always fall back to downloading the full update. Our shortcut is still created as `GitHub Desktop`. The exe name change has an unfortunate side effect: the first update where we make this change, "restart now" won't work as Squirrel will still be looking for the old exe name. I think that's OK given that it's a one-time problem for a beta.
3. I tried updating `electron-winstaller` to get the GIF RAM usage fix, but there's a bug in Squirrel.Windows with creating diffs for some of our files (https://github.com/Squirrel/Squirrel.Windows/issues/1101).

#### Testing

I tested this using the new beta channel:

1. Created a new beta release.
1. Verified the delta is uploaded to Central and is smaller than the full release.
1. Installed the beta build.
1. Created another beta release.
1. Verified the delta is uploaded to Central and is smaller than the full release.
1. Check for updates with the installed beta.
1. Verify that the installed beta updates to the new beta using the delta, via the logging in `SquirrelSetup.log`.
